File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(string number, int k) {
    string ans = "";
    int cnt = 0;
    for(char num : number) {
        while(cnt < k && ans.size() && ans.back() < num) {
            cnt++;
            ans.pop_back();
        }
        ans.push_back(num);
    }
    while(cnt < k && ans.size()) {
        ans.pop_back();
        cnt++;
    }
    return ans;
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var stk = [Character]()
    var nums = Array(number)
    var remaining = k
    
    for num in nums {
        while !stk.isEmpty && remaining > 0 && stk.last! < num {
            remaining -= 1
            _ = stk.removeLast()
        }
        stk.append(num)
    }
    
    while remaining > 0 {
        _ = stk.removeLast()
        remaining -= 1
    }
    return String(stk)
}